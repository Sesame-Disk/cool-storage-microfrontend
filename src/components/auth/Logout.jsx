import { useContext } from "react";
import { useNavigate } from "react-router-dom";
import AuthContext from "../../store/auth-context";

const Logout = (props) => {
  const authContext = useContext(AuthContext);
  const navigate = useNavigate();
  const handleLogout = () => {
    authContext.logout();
    navigate("/");
  };
  return (
    <button className={props.className} onClick={handleLogout}>
      Logout
    </button>
  );
};

export default Logout;
