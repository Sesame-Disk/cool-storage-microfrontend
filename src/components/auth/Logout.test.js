import React from "react";
import renderer from "react-test-renderer";
import { render, screen } from "@testing-library/react";
import Logout from "./Logout";

jest.mock("react-router-dom", () => ({
  useNavigate: jest.fn(),
}));

describe("Logout", () => {
  it("renders without crashing", () => {
    render(<Logout />);
    expect(screen.getByText(/logout/i)).toBeInTheDocument();
  });
  it("renders correctly as espected", () => {
    const tree = renderer.create(<Logout />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
