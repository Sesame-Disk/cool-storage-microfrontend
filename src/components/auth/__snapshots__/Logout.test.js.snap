// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout renders correctly as espected 1`] = `
<button
  onClick={[Function]}
>
  Logout
</button>
`;
